internal/access/grant.go

```diff
@@ -30,7 +30,15 @@ func ListGrants(c *gin.Context, subject uid.PolymorphicID, resource string, priv
 	roles := []string{models.InfraAdminRole, models.InfraViewRole, models.InfraConnectorRole}
 	db, err := RequireInfraRole(c, roles...)
 	if err == nil {
-		selectors = append(selectors, data.ByOptionalSubject(subject))
+		if inherited && subject.IsIdentity() {
```

Comment:
I feel like a user should be able to request their own inherited roles too

Reply:
Not sure what you're talking about here. you're talking about grants that grant access to roles, then those roles have their own grants? We don't have any of those today, but I agree with you.

Reply:
I mean the case like this:
```
if subject.IsGroup() && userBelongsToGroup(user, subject) {
  // add selector for this group
}
```

I don't think users could list grants for groups they belong to with this auth check?

Reply:
I don't think `inherited` means anything if the query is for a group. A group can't inherit grants, only a user inherits grants from a group.

But I think there is a case missing here. We're only using this flag when the request is authorized by a grant. The block below (where the request is authorized by authn identity matching the query) is ignoring `inherited`.

Reply:
I added a test for this and it works as is

Reply:
If you mean `TestListGrants`, I think that is misleading. It's being authorized by the group grant, not by identity.

I'm pretty sure the scenario I described does not work.